iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/plugin/KubernetesParser.java

```diff
@@ -102,6 +102,21 @@ private FileTree parseHelmFile(String source, @Nullable HelmInputFileContext inp
     return result;
   }
 
+  private Optional<FileTree> validateInputFileContext(@Nullable HelmInputFileContext inputFileContext) {
+    if (inputFileContext == null) {
+      LOG.debug("No InputFileContext provided, skipping processing of Helm file");
+      return Optional.ofNullable(super.parse("{}", null, FileTree.Template.HELM));
+    }
+
+    if (("values.yaml".equals(inputFileContext.inputFile.filename()) || "values.yml".equals(inputFileContext.inputFile.filename())) &&
+      new KubernetesSensor.HelmProjectMemberPredicate(inputFileContext.sensorContext).apply(inputFileContext.inputFile)) {
```

Comment:
Do we really need this check here? We only get into this method if a file satisfies Sensor's predicates and has Helm content. I'd rather suggest to check that this file is in the root of chart directory, not to skip things like `templates/values.yaml`